### PR TITLE
Enable task tests

### PR DESCRIPTION
**Pull request by cdmihai**
Enabled tests for the tasks that go into netcore.
Skiped the ones that fail and documented them with GH issues.

Unfinished test files due to hundreds of failing tests (will study each one individually). I left them out of the build for now:
- GenerateResource_Tests
- ResolveAssemblyReference_Tests

### CODE REVIEW

File: src/XMakeTasks/UnitTests/Microsoft.Build.Tasks.UnitTests.csproj [hunk 1]
```diff
@@ -67,6 +67,16 @@
...
+    <Compile Include="AssignTargetPath_Tests.cs" />
+    <Compile Include="ConvertToAbsolutePath_Tests.cs" />
+    <Compile Include="CreateCSharpManifestResourceName_Tests.cs" />
+    <Compile Include="FindAppConfigFile_Tests.cs" />
+    <Compile Include="FindUnderPath_Tests.cs" />
+    <Compile Include="GenerateBindingRedirects_Tests.cs" />
+    <Compile Include="GetReferencePaths_Tests.cs" />
+    <Compile Include="ResolveCodeAnalysisRuleSet_Tests.cs" />
+    <Compile Include="ToolTaskExtension_Tests.cs" />
+
```
Comment (rainersigwald) [inline src/XMakeTasks/UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]: Can you preserve alphabetical order in this list for hopefully-easier future merging?

**Comment by rainersigwald** [comment]
LGTM.  Don't dive into ResolveAssemblyReferences before talking it over first; that one will probably be A Lot Of Fun.

File: src/XMakeTasks/UnitTests/Microsoft.Build.Tasks.UnitTests.csproj (hunk 1 above)
Comment (cdmihai) [inline src/XMakeTasks/UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]: Done. Thanks.

Mihai

**Comment by AndyGerlicher** [comment]
LGTM

**Pull request merged**